src/tools/remote/src/test/java/com/google/devtools/build/remote/worker/HttpCacheServerHandlerTest.java

```diff
@@ -0,0 +1,38 @@
+package com.google.devtools.build.remote.worker;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class HttpCacheServerHandlerTest {
+
+  private HttpCacheServerHandler handler = new HttpCacheServerHandler();
+
+  @Test
+  public void isUriValidWhenValidUri() {
+    assertTrue(handler.isUriValid("http://some-path.co.uk:8080/ac/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("http://127.12.12.0:8080/ac/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("http://localhost:8080/ac/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("https://localhost:8080/ac/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("localhost:8080/ac/1111111111111111111111111111111111111111111111111111111111111111"));
+
+    assertTrue(handler.isUriValid("http://some-path.co.uk:8080/cas/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("http://127.12.12.0:8080/cas/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("http://localhost:8080/cas/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("https://localhost:8080/cas/1111111111111111111111111111111111111111111111111111111111111111"));
+    assertTrue(handler.isUriValid("localhost:8080/cas/1111111111111111111111111111111111111111111111111111111111111111"));
+  }
```

Comment:
IIRC `HttpRequest.getUri()` for servers will return the `path`. So maybe test `cas/hash` too.